scripts/apps/dashboard/user-activity/components/UserActivityWidget.tsx

```diff
@@ -0,0 +1,411 @@
+                            {...getQueryNotMarkedOrMarkedForMe(userId)},
+                            {...getQueryNotLockedOrLockedByMe(userId)},
+                        ],
+                    },
+                }).then((res) => {
+                    res._items = res._items.filter(filterOutItemsInIncomingStage(services));
+
+                    return res;
+                });
+            },
+        },
+        {
+            id: 'moved',
+            label: gettext('Moved to a working stage by this user'),
+            query(fetchFn) {
+                return fetchFn('archive_history', {...getQueryMovedByUser(userId)})
+                    .then((res) => {
+                        res._items = res._items.filter(onlyHistoryItemsInWorkingStage(services));
+
+                        return res;
+                    });
+            },
+        },
+    ];
+};
+
+function onlyHistoryItemsInWorkingStage({desks}) {
+    return (historyItem) => {
+        const stage = getStageForItem(historyItem, {desks});
+
+        return stage.working_stage === true;
+    };
+}
+
+function filterOutItemsInIncomingStage({desks}) {
+    return (item) => {
+        const stage = getStageForItem(item, {desks});
+
+        return stage.default_incoming === false;
+    };
+}
+
+function getStageForItem(item, {desks}) {
+    const stageId = item.task?.stage;
+
+    if (!stageId) {
+        return false;
+    }
+
+    const stage = desks.stageLookup[stageId] || null;
+
+    if (!stage) {
+        console.warn('Tried to find a stage with an invalid id', stageId);
+    }
+
+    return stage;
+}
+
+export default class UserActivityWidget extends React.Component<{}, IState> {
+    services: any;
+
+    constructor(props) {
+        super(props);
+
+        this.services = {
+            $anchorScroll: ng.get('$anchorScroll'),
+            $location: ng.get('$location'),
+            $q: ng.get('$q'),
+            $rootScope: ng.get('$rootScope'),
+            $timeout: ng.get('$timeout'),
+            activityService: ng.get('activityService'),
+            api: ng.get('api'),
+            authoringWorkspace: ng.get(
+                'authoringWorkspace',
+            ) as AuthoringWorkspaceService,
+            cards: ng.get('cards'),
+            datetime: ng.get('datetime'),
+            desks: ng.get('desks'),
+            metadata: ng.get('metadata'),
+            search: ng.get('search'),
+            superdesk: ng.get('superdesk'),
+        };
+
+        this.state = {
+            userSearchField: '',
+            searchOpen: true,
+            groups: null,
+            groupsData: null,
+            loading: false,
+        };
+    }
+
+    componentDidMount() {
+        [
+            'item:lock',
+            'item:unlock',
+            'item:spike',
+            'item:move',
+            'item:publish',
+            'item:update',
+        ].map((event) =>
+            this.services.$rootScope.$on(event, this.refreshItems.bind(this)),
```

Comment:
I thought about it, but not sure how to do it. Would this work?

```ts
$rootScope.$off(event, this.refreshItems)
```

Reply:
[screenshot_18](https://user-images.githubusercontent.com/2076953/72790442-e3bd7900-3c35-11ea-9bdd-db8d481570d1.png) yep. You need to do it the same way for every event, except you have to move out the binding because biding creates a new function reference and removing existing listeners wouldn't work

Reply:
woops, I forgot it's an angular scope, not js events. I've looked at the docs and didn't see $off method. Looks like calling $on returns a function which you need to store and call when you want to remove the listener https://stackoverflow.com/a/14898795/1175593